test/e2e/elasticsearch_test.go

```diff
@@ -113,104 +134,192 @@ func (suite *ElasticSearchTestSuite) TestSimpleProd() {
 func (suite *ElasticSearchTestSuite) TestEsIndexCleanerWithIndexPrefix() {
-	esIndexCleanerEnabled = false
-	esIndexPrefix := "prefix"
-	jaegerInstanceName := "test-es-index-prefixes"
-	jaegerInstance := &v1.Jaeger{}
-
-	if skipESExternal {
-		esNamespace = namespace
-		numberOfDays := 0
-		indexCleanerSpec := v1.JaegerEsIndexCleanerSpec{
-			Enabled:      &esIndexCleanerEnabled,
-			Schedule:     "*/1 * * * *",
-			NumberOfDays: &numberOfDays,
-		}
-
-		jaegerInstance = getJaegerSelfProvSimpleProd(jaegerInstanceName, namespace, 1)
-		jaegerInstance.Spec.Storage.EsIndexCleaner = indexCleanerSpec
-		addIndexPrefix(jaegerInstance, esIndexPrefix)
-
-		createESSelfProvDeployment(jaegerInstance, jaegerInstanceName, namespace)
-		defer undeployJaegerInstance(jaegerInstance)
-
-		ProductionSmokeTest(jaegerInstanceName)
-	} else {
-		esNamespace = storageNamespace
-		jaegerInstance = getJaegerAllInOne(jaegerInstanceName)
-		addIndexPrefix(jaegerInstance, esIndexPrefix)
+	suite.runIndexCleaner("my-custom_prefix", []int{3, 1, 0})
+}
 
-		err := fw.Client.Create(context.Background(), jaegerInstance, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
-		require.NoError(t, err, "Error deploying Jaeger")
-		defer undeployJaegerInstance(jaegerInstance)
-		err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout)
-		require.NoError(t, err, "Error waiting for deployment")
+// executes es index cleaner with default index prefix
+func (suite *ElasticSearchTestSuite) TestEsIndexCleaner() {
+	suite.runIndexCleaner("", []int{45, 30, 7, 1, 0})
+}
 
-		// Run the smoke test so indices will be created
-		AllInOneSmokeTest(jaegerInstanceName)
+// executes index cleaner tests
+func (suite *ElasticSearchTestSuite) runIndexCleaner(esIndexPrefix string, daysRange []int) {
+	logrus.Infof("index cleaner test started. daysRange=%v, prefix=%s", daysRange, esIndexPrefix)
+	jaegerInstanceName := "test-es-index-cleaner"
+	if esIndexPrefix != "" {
+		jaegerInstanceName = "test-es-index-cleaner-with-prefix"
 	}
-	// Now verify that we have indices with the prefix we want
-	indexWithPrefixExists(esIndexPrefix+"-jaeger-", true, esNamespace)
-
-	// Turn on index clean and make sure we clean up
-	turnOnEsIndexCleaner(jaegerInstance)
-	indexWithPrefixExists(esIndexPrefix+"-jaeger-", false, esNamespace)
+	// get jaeger CR to create jaeger services
+	jaegerInstance := getJaegerSelfProvSimpleProd(jaegerInstanceName, namespace, 1)
 
-}
+	// storage namespace
+	esNamespace = namespace
 
-func addIndexPrefix(jaegerInstance *v1.Jaeger, esIndexPrefix string) {
-	// Add an index prefix to the CR before creating this Jaeger instance
-	options := jaegerInstance.Spec.Storage.Options.Map()
-	updateOptions := make(map[string]interface{})
-	for key, value := range options {
-		updateOptions[key] = value
+	// update es node namespace and es node url into jaeger CR for external es deployment
```

Comment:
This comment doesn't really make it clear what your doing here.  I think something like this might be better:
// If there is an external es deployment use it instead of creating a self prov one